src/Nest/Aggregations/Bucket/Bucket.cs

```diff
@@ -1,35 +1,50 @@
+using System;
 using System.Collections.Generic;
 
 namespace Nest
 {
-	public interface IBucket : IAggregation
+
+	public class Bucket : AggregationsHelper, IAggregationResult
 	{
-		IDictionary<string, IAggregation> Aggregations { get; }
+		public Bucket() { }
+		public Bucket(IDictionary<string, IAggregationItem> aggregations) : base(aggregations) { }
+
+		public IDictionary<string, object> Meta { get; set; }
 	}
 
-	public interface IBucketItem : IAggregation
+	public class Bucket<TBucketItem> : Bucket
+		where TBucketItem : IBucketItem
 	{
+		public Bucket() { }
+		public Bucket(IDictionary<string, IAggregationItem> aggregations) : base(aggregations) { }
+
+		public IList<TBucketItem> Items { get; set; }
 	}
 
-	public class Bucket : IAggregation
+	public class DocCountBucket : Bucket
 	{
-		public IEnumerable<IAggregation> Items { get; set; }
-		public long? DocCountErrorUpperBound { get; set; }
-		public long? SumOtherDocCount { get; set; }
+		public DocCountBucket() { }
+		public DocCountBucket(IDictionary<string, IAggregationItem> aggregations) : base(aggregations) { }
+
+		public long DocCount { get; internal set; }
 	}
 
-	public class Bucket<TBucketItem> : BucketBase
+	public class DocCountBucket<TBucketItem> : DocCountBucket
 		where TBucketItem : IBucketItem
 	{
-		public Bucket() { }
-		public Bucket(IDictionary<string, IAggregation> aggregations) : base(aggregations) { }
+		public DocCountBucket() { }
+		public DocCountBucket(IDictionary<string, IAggregationItem> aggregations) : base(aggregations) { }
 
 		public IList<TBucketItem> Items { get; set; }
 	}
 
-	public abstract class BucketBase : AggregationsHelper, IBucket
+	// Intermediate object used for deserialization
+	internal class BucketDto : IAggregationResult
```

Comment:
I'm really hating DTO as a suffix but can not suggest a more meaningful name for this :smile: